Refactor Test class to allow for canRun checks

desbo opened the pull request:
In addition to audience segmentation, you can now create tests with or without can run criteria, which I've called "split tests" and "conditional tests". The can run check is a boolean function that takes a request as its only parameter. I figure this should be enough for most use cases.

@guardian/contributions

desbo committed:
Refactor Test class to allow for canRun checks

In addition to audience segmentation, you can now create tests with or without can run criteria, which I've
called "split tests" and "conditional tests". The can run check is a
boolean function that takes a request as its only parameter. I figure
this should be enough for most use cases.

alexduf commented on app/abtests/Test.scala:
```diff
@@ -2,26 +2,42 @@ package abtests
   }
 }
 
+case class SplitTest(name: String, audienceSize: Percentage, audienceOffset: Percentage, variants: Variant*) extends Test {
```
It looks like you can keep the `Test` `case class` and define `canRun` as a function with a default value

alexduf reviewed (commented):
Looks good otherwise 👍

desbo committed:
Use single Test case class

desbo committed:
Add "force into test" functionality for server-side tests

This requires using a querystring rather than an anchor link, as anchors
aren't passed through in the request.

desbo committed:
Fix build
